### PR TITLE
Add a regression test for #1280

### PR DESCRIPTION
**Pull request by banach-space**

### CODE REVIEW

File: flang/test/Driver/emit-llvm-all-drivers.f90 [hunk 1]
```diff
@@ -0,0 +1,22 @@
+! RUN: %flang_fc1 -emit-llvm %s -o %t.flang | FileCheck %s
+! RUN: %flang_fc1 -emit-mlir %s -o %t.flang_tco | tco | FileCheck %s
+! RUN: bbc %s -o - | tco -o %t.bbc_tco | FileCheck %s
+
+! This is added as a regression test for a failure reported in https://github.com/flang-compiler/f18-llvm-project/issues/1280. It
+! makes sure that this example can be lowered to LLVM IR by all major drivers in Flang. Ultimately, we should be veryfying that the
+! generated code is identical in all the tested cases. Currently, there are some discrepancy in the the generated debug info.
+! TODO: Make sure that the generated code is identical in all cases
```
Comment (clementval) [inline flang/test/Driver/emit-llvm-all-drivers.f90:8]: Looks like more than 80.

Comment (banach-space) [reply flang/test/Driver/emit-llvm-all-drivers.f90:8]: It is. Don’t mind updating, though the 80 char limit is rarely applied to test files.

Comment (clementval) [reply flang/test/Driver/emit-llvm-all-drivers.f90:8]: Up to you.

Comment (banach-space) [reply flang/test/Driver/emit-llvm-all-drivers.f90]: Updated the comment.

Also, fixed the test :) (sorry, was messing around and forgot to restore the original version before sending to GitHub - should be all good now!)

File: flang/test/Driver/emit-llvm-all-drivers.f90 [hunk 2]
```diff
@@ -0,0 +1,25 @@
+! RUN: %flang_fc1 -emit-llvm %s -o - | FileCheck %s
+! RUN: %flang_fc1 -emit-mlir %s -o - | tco | FileCheck %s
+! RUN: bbc %s -o - | tco -o - | FileCheck %s
```
Comment (kiranchandramohan) [inline flang/test/Driver/emit-llvm-all-drivers.f90:3]: Nit: The tco invocations look different.

File: flang/test/Driver/emit-llvm-all-drivers.f90 [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+! This is added as a regression test for a failure reported in
+! https://github.com/flang-compiler/f18-llvm-project/issues/1280. It makes sure
+! that this example can be lowered to LLVM IR by all major drivers in Flang.
+! Ultimately, we should be veryfying that the generated code is identical in
+! all the tested cases. Currently, there are some discrepancy in the the
+! generated debug info.
```
Comment (kiranchandramohan) [inline flang/test/Driver/emit-llvm-all-drivers.f90:10]: Nit: there `is` some discrepancy in `the` generated
or
Nit: there `are` some discrepancies in `the` generated

**Review by clementval — APPROVED** [review]

**Review by kiranchandramohan — APPROVED** [review]
Looks good to me.

**Comment by banach-space** [comment]
I'll address Kiran's comments and squash this. Thanks for taking a look!

**Review by schweitzpgi — APPROVED** [review]
Thanks for adding this.

It might make sense to add a `Regression/` directory under `test/` since this isn't exclusively a driver test.

**Comment by banach-space** [comment]
> It might make sense to add a `Regression/` directory under `test/`

This might be a bit confusing. AFAIK, all LIT tests in LLVM are considered as [Regression tests](https://llvm.org/docs/TestingGuide.html#regression-tests). Also, I'm not aware of any other sub-project having a dedicated directory specifically for regressions. 

> this isn't exclusively a driver test.

True. But that's the case for most tests in "flang/test/Driver". The driver integrates various parts of Flang and most tests just verify this integration. So perhaps it's not that bad?

**Comment by schweitzpgi** [comment]
> > It might make sense to add a `Regression/` directory under `test/`
> 
> This might be a bit confusing. AFAIK, all LIT tests in LLVM are considered as [Regression tests](https://llvm.org/docs/TestingGuide.html#regression-tests). Also, I'm not aware of any other sub-project having a dedicated directory specifically for regressions.

The name wasn't the focal point. Putting tests in the driver area that are not meant to test the driver is confusing.

**Comment by banach-space** [comment]
> The name wasn't the focal point. Putting tests in the driver area that are not meant to test the driver is confusing.

How about "test/driver/compatibility"? "test/driver/bridge-compatibility"? "test/driver/driver-bridge-compatibility"?

**Pull request merged**